### ISSUE TITLE
KDE Widget not working - Must run mycroft.sh start manually for it to work.

### ISSUE THREAD

**Issue by chrison999** [MycroftAI/mycroft-core#757]
Several people have reported that the KDE Plasmoid widget shows a "Connection Error" upon trying to use the widget.  I did some testing and found a couple of things:

1.  Upon first using a terminal program and running "~/mycroft-core/mycroft.sh start" I get an error message saying there's a permission problem with /opt/mycroft/skills.  I'm then prompted for my password and the problem is fixed.

2.  After running "~/mycroft-core/mycroft.sh start" and the clicking the "run/stop" button at the top-left of the widget a couple of times, mycroft then runs from the widget just fine.  However, if I do a reboot of KDE/Plasma I then have to once again manually repeat running "mycroft.sh start" in a terminal to get the widget working again.

**Comment by AIIX** [comment]
chrison999,  I have not been able to replicate the error message saying there's a permission problem with /opt/mycroft/skills and being prompted for a pass. So i am not sure how this affects the plasmoid connection issue.

On the 2nd point:  As per my testing the plasmoid on a KDE Neon latest user edition in Virtual Box environment the widget runs without connection errors except at times when the connection error seems due an issue with how long the plasmoid takes to connect to the messagebus, the following commit: https://cgit.kde.org/plasma-mycroft.git/commit/?id=9eeffba2770731d037f5e32e8274e498ed6f62a8  should increase the delay before the plasmoid tries connecting to mycroft.  Additionally also try it on the latest version of KDE Neon.

**Comment by chrison999** [comment]
The version of KDE Neon I used to create the vm on vbox was this:

neon-useredition-20170504-1018-amd64

However, I'll see if there's a newer version available and give that a try.  I'll also try it in a docker container and see if the problem is the same there.

**Comment by chrison999** [comment]
The iso I used to create the vm on vbox was neon-useredition-20170504-1018-amd64.  I checked and that's the latest release of the user edition.  There is a newer version of the lts release (neon-userltsedition-20170517-0019-amd64.iso) so I'm downloading that and I'll give it a try to see if the same problem occurs.

As well, I took a look at the commit.  Increasing the delay before mycroft tries to connect to the messagebus will probably help.

**Comment by chrison999** [comment]
I created a vbox vm with the latest lts edition (as noted above) and got the same result...  widget reports "Connection Error" until I manually start mycroft in a Konsole terminal window (using "mycroft.sh start") and then click the on/off button a few times to get it to connect.  Also, I got the "permission error with /opt/mycroft folder" again when I ran mycroft.sh start.

I couldn't try the widget using a KDE Neon docker image because the widget's install script requires a reboot and as soon as I did that the install was lost.  (Not sure how to keep the drive data persistent in docker???)

**Comment by AIIX** [comment]
Can you post a plasmashell log for when you press the start button. Does mycroft start up in the background and give you the audio message skills updated or can you ask it anything even if the widget shows connection error ? Voice should work regardless of the plasmoid if mycroft is starting up in the background.

**Comment by AIIX** [comment]
Additionally if mycroft does not start in the background try the following:

Go to folder: /usr/share/plasma/plasmoids/org.kde.plasma.mycroftplasmoid/contents/code/
Open file: startservice.sh
Replace: "./mycroft.sh start" with: "./start.sh service && ./start.sh skills && ./start.sh voice"
Logout and Re-login and check if mycroft does start.

**Comment by chrison999** [comment]
Where can I find the plasmashell log?

When I first log into KDE/Plasma, nothing is started in the background.  I did "ps" commands from Konsole as both root and my username and found nothing loaded.  See attached screenshot:

![screenshot from 2017-05-19 10-26-45](https://cloud.githubusercontent.com/assets/7979787/26260599/a56dd43e-3c82-11e7-953d-c08284eeaca3.png)

I changed the startservice.sh file as you requested and rebooted.  This time, I got the "Mycroft is Ready" response.  Typing a command gets the command in the history but no response from mycroft (neither the command history nor audio).  So, I opened a terminal again and checked.  The service module was loaded but nothing else.

Next, I killed the service module and tried "./start.sh service && ./start.sh skills && ./start.sh voice" from the command line.  Mycroft loaded the service module and then stopped.  Skills and voice weren't loaded.  I waited around 10 minutes to see if the script would continue but it didn't.

So, to summarize, with "mycroft.sh start" in the startservice.sh script nothing gets started when I log into KDE Neon.  With "./start.sh service && ./start.sh skills && ./start.sh voice" in the startservice.sh script, the service module gets loaded but nothing else.  This was confirmed by running "./start.sh service && ./start.sh skills && ./start.sh voice" from a command line.

Hope that helps a bit.  If you can tell me where to find the plasmashell log file I'll check that for you.

**Comment by AIIX** [comment]
The above was a error on my end it should have been "/start.sh service & ./start.sh skills & ./start.sh voice" rather than "&&". on the permissions part id suggest first do: "sudo chmod -R 777 /opt/mycroft/skills/"  

To get debug log: "plasmoidviewer -a org.kde.plasma.mycroftplasmoid"

**Comment by chrison999** [comment]
Ok, changing "&&" to &" worked although I had to press the on/off button a few times to get the "Mycroft is Ready" message.

**Comment by forslund** [comment]
Hi, was this resolved to satisfaction? Can the issue be closed?

**Comment by forslund** [comment]
Since there's been no communication in a long time I'm closing this issue.

**Comment by DrMemoryFish** [comment]
> org.kde.plasma.mycroftplasmoid

org.kde.plasma.mycroftplasmoid file doesnt exist after installing, how do i fix this?
